src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor() {
+    this.health = 100;
+    this.hidden = false;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+
+  bite(animal) {
+    if (!animal.hidden && !(animal instanceof Carnivore)) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
```

Comment:
```suggestion
    if (animal.health === 0) {
```
Меньше нуля он не должен стать, ни в каком случае

Reply:
Ok